server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -158,6 +162,108 @@ async fn list_attempted_messages(
     )))
 }
 
+/// Additional parameters (besides pagination) in the query string for the "List Attempts by
+/// Endpoint" enpoint.
+#[derive(Debug, Deserialize, Validate)]
+pub struct ListAttemptsByEndpointQueryParameters {
+    status: Option<MessageStatus>,
+    status_code_class: Option<StatusCodeClass>,
+    #[validate]
+    event_types: Option<EventTypeNameSet>,
+    #[validate]
+    channel: Option<EventChannel>,
+}
+
+/// Fetches a list of [`MessageAttemptOut`]s for a given endpoint ID
+async fn list_attempts_by_endpoint(
+    Extension(ref db): Extension<DatabaseConnection>,
+    ValidatedQuery(mut pagination): ValidatedQuery<Pagination<MessageAttemptId>>,
+    ValidatedQuery(ListAttemptsByEndpointQueryParameters {
+        status,
+        status_code_class,
+        event_types,
+        channel,
+    }): ValidatedQuery<ListAttemptsByEndpointQueryParameters>,
+    Path((_app_id, endp_id)): Path<(ApplicationId, EndpointId)>,
+    AuthenticatedApplication {
+        permissions: _,
+        app,
+    }: AuthenticatedApplication,
+) -> Result<Json<ListResponse<MessageAttemptOut>>> {
+    let limit = pagination.limit;
+    let iterator = pagination.iterator.take();
+
+    // Confirm endpoint ID belongs to the given application
+    if endpoint::Entity::secure_find_by_id(app.id, endp_id.clone())
+        .one(db)
+        .await?
+        .is_none()
+    {
+        return Err(Error::Http(HttpError::not_found(None, None)));
+    }
+
+    let mut query = messageattempt::Entity::secure_find_by_endpoint(endp_id)
+        .order_by_desc(messageattempt::Column::Id)
+        .limit(limit + 1);
+
+    if let Some(iterator) = iterator {
+        query = query.filter(messageattempt::Column::Id.lt(iterator));
+    }
+
+    if let Some(status) = status {
+        query = query.filter(messageattempt::Column::Status.eq(status));
+    }
+
+    query = match status_code_class {
+        Some(StatusCodeClass::Code1xx) => {
+            query.filter(messageattempt::Column::ResponseStatusCode.lt(100i16))
+        }
+
+        Some(StatusCodeClass::Code2xx) => {
+            query.filter(messageattempt::Column::ResponseStatusCode.between(199i16, 300i16))
```

Comment:
Do we really need to explicitly mark it as i16? Doesn't rust do it automatically?